core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -355,6 +361,22 @@ public void removeBlock(long sessionId, long blockId)
     mBlockStore.removeBlock(sessionId, blockId);
   }
 
+  // TODO(Tony Sun): Currently no data access, locks needed or not?
+  public void freeWorker() throws IOException{
+    List<StorageTier> curTiers = IntStream.range(0, WORKER_STORAGE_TIER_ASSOC.size()).mapToObj(
+                    tierOrdinal -> DefaultStorageTier.newStorageTier(
+                            WORKER_STORAGE_TIER_ASSOC.getAlias(tierOrdinal),
+                            tierOrdinal,
+                            WORKER_STORAGE_TIER_ASSOC.size() > 1))
+            .collect(toImmutableList());
+    for (StorageTier tier : curTiers) {
+      for (StorageDir dir : tier.getStorageDirs())  {
+        FileUtils.deletePathRecursively(dir.getDirPath());
```

Comment:
If some directory fails to be deleted, then the rest directories will be skipped form deletion

Reply:
Fixed